### PR TITLE
chore(project): initial commit

### PR DESCRIPTION
**Pull request by VMBindraban**
Doesn't include support for multiple `equals` for the same property. You can use `in` for that. 
Nice feature to have in the future.

### CODE REVIEW

File: src/criteriaBuilder.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          return;
+        }
+
+        filtersParsed = extend(filtersParsed, this.parseField(filter.data.field, this.parseOperator(filter.data)));
+      });
+
+      blocks.push(filtersParsed);
+    });
+
+    this.criteria = (blocks.length > 1) ? {where: {or: blocks}} : {where: blocks[0]};
```
Comment (doktordirk) [inline src/criteriaBuilder.js]: that's sails specific. maybe it can be 'outsources' to allow different formats

File: src/criteriaBuilder.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        return this.toArray(filter.value);
+      case 'not-in':
+        filter.operator = 'not';
+        filter.value    = this.toArray(filter.value);
+      default:
+        return {[filter.operator]: filter.value};
+    }
+  }
+
+  parseBetween(data, valueOperator = '>=', betweenOperator = '<=') {
```
Comment (doktordirk) [inline src/criteriaBuilder.js:45]: same here i guess. would be nice for other to have it configurable as the defaults are used above

File: src/criteriaBuilder.js (hunk 1 above)
Comment (RWOverdijk) [inline src/criteriaBuilder.js]: Oof. Difficult. We could create criteria builders for backends I suppose. If so, I do vote for _only_ building one for sails. Support for others can be added by those who need it.

Comment (doktordirk) [inline src/criteriaBuilder.js]: @Rw i.d suggest to generalize the format and to move the logic to orm or api. then users can install aurelia-api-hapi-adapter and be done with it for all our plugins

**Pull request merged**